functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianAbstractTestClass.java

```diff
@@ -0,0 +1,139 @@
+  private TestWorkspace workspace;
+
+  @Inject private CodenvyEditor editor;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private TestApiEndpointUrlProvider testApiEndpointUrlProvider;
+
+  private Integer EXPECTED_ERROR_LINE;
+  private Integer INJECTION_ENTRY_POINT;
+  private String EXPECTED_ERROR_TEXT;
+  private String PROJECT_FILE;
+  private String PATH_TO_FILE;
+  private String PROJECT_NAME;
+  private String PROJECT_DEPENDENCY;
+  private String ERROR_MESSAGE;
+  private static final String CHE_PROD_PREVIEW_URL = "che.prod-preview.openshift.io";
+
+  public void setExpectedErrorLine(Integer expectedErrorLine) {
+    EXPECTED_ERROR_LINE = expectedErrorLine;
+  }
+
+  public void setInjectionEntryPoint(Integer injectionEntryPoint) {
+    INJECTION_ENTRY_POINT = injectionEntryPoint;
+  }
+
+  public void setExpectedErrorText(String expectedErrorText) {
+    EXPECTED_ERROR_TEXT = expectedErrorText;
+  }
+
+  public void setProjectFile(String projectFile) {
+    PROJECT_FILE = projectFile;
+  }
+
+  public void setPathToFile(String pathToFile) {
+    PATH_TO_FILE = pathToFile;
+  }
+
+  public void setProjectName(String projectName) {
+    PROJECT_NAME = projectName;
+  }
+
+  public void setProjectDependency(String projectDependency) {
+    PROJECT_DEPENDENCY = projectDependency;
+  }
+
+  public void setErrorMessage(String errorMessage) {
+    ERROR_MESSAGE = errorMessage;
+  }
+
+  public void setWorksapce(TestWorkspace workspace) {
+    this.workspace = workspace;
+  }
+
+  public void openTestFile() throws Exception {
+    checkWorkspace(workspace);
+    projectExplorer.waitItem(PROJECT_NAME);
+  }
+
+  @BeforeMethod
+  public void prepareProjectFile() {
+    projectExplorer.expandPathInProjectExplorerAndOpenFile(PATH_TO_FILE, PROJECT_FILE);
+    appendDependency();
+  }
+
+  @AfterMethod
+  public void closeFiles() {
+    removeDependency();
+    closeFilesAndProject();
+  }
+
+  @AfterClass
+  public void closeBrowser() {
+    super.closeBrowser();
+  }
+
+  @Test
+  public void checkBayesianError() {
+    editorCheckBayesianError();
+    LOG.info("Bayesian error message was present after adding dependency.");
+  }
+
+  @Test
+  public void checkErrorPresentAfterReopenFile() {
+    closeFilesAndProject();
+    projectExplorer.expandPathInProjectExplorerAndOpenFile(PATH_TO_FILE, PROJECT_FILE);
+    editorCheckBayesianError();
+    LOG.info("Bayesian error message was present after reopening file.");
+  }
+
+  protected void appendDependency() {
+    editor.setCursorToLine(INJECTION_ENTRY_POINT);
+    editor.typeTextIntoEditor(PROJECT_DEPENDENCY);
+    editor.waitTabFileWithSavedStatus(PROJECT_FILE);
+  }
+
+  protected void removeDependency() {
+    editor.setCursorToLine(INJECTION_ENTRY_POINT);
+    editor.deleteCurrentLine();
+    editor.waitTabFileWithSavedStatus(PROJECT_FILE);
+  }
+
+  protected void editorCheckBayesianError() {
+    if (testApiEndpointUrlProvider.get().getHost().equals(CHE_PROD_PREVIEW_URL)) {
```

Comment:
I don't think this is the right approach... In this case, when bayesian is fixed in prod-preview, this test will be still skipped...
IMHO we want to skip the test only if it fails AND the env is prod-preview.